common/src/main/java/tachyon/heartbeat/HeartbeatScheduler.java

```diff
@@ -63,19 +66,23 @@ public static void addTimer(ScheduledTimer timer) {
   public static synchronized void removeTimer(ScheduledTimer timer) {
     Preconditions.checkNotNull(timer);
     sLock.lock();
-    sTimers.remove(timer.getThreadName());
-    sLock.unlock();
+    try {
+      sTimers.remove(timer.getThreadName());
+    } finally {
+      sLock.unlock();
+    }
```

Comment:
The advantages I see are 
1) There's a low but nonzero possibility that we won't want to crash the program on an unchecked or [sneakily thrown](https://projectlombok.org/features/SneakyThrows.html) checked exception
2) The code could be changed in the future so that a handleable exception can be thrown 
3) This will make Findbugs happy, which will help us find situations where we actually do have a concurrency error 
4) This is an appeal to authority, but it's generally considered good practice to follow any locking with `try...finally`: http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/locks/Lock.html